packages/bcp/src/cosmwasmconnection.ts

```diff
@@ -372,9 +372,12 @@ export class CosmWasmConnection implements BlockchainConnection {
 
     // tslint:disable-next-line: deprecation
     const accountForHeight = await this.restClient.authAccounts(senderAddress, response.height);
-    // this is technically not the proper nonce. maybe this causes issues for sig validation?
+    const accountNumber = accountForHeight.result.value.account_number;
```

Comment:
Nice. I didn't realize there was a query here already, totally makes sense to handle this properly if we have access to the info

Reply:
I still don't know how reliable the sequence is. But we don't do verification, so it is not critical. At least the code is more expressive now.